providers/flagsmith/src/main/java/dev.openfeature.contrib.providers.flagsmith/FlagsmithProvider.java

```diff
@@ -0,0 +1,232 @@
+        String key, Integer defaultValue, EvaluationContext evaluationContext) {
+        return resolveFlagsmithEvaluation(key, defaultValue, evaluationContext, Integer.class);
+    }
+
+    @Override
+    public ProviderEvaluation<Double> getDoubleEvaluation(
+        String key, Double defaultValue, EvaluationContext evaluationContext) {
+        return resolveFlagsmithEvaluation(key, defaultValue, evaluationContext, Double.class);
+    }
+
+    @Override
+    public ProviderEvaluation<Value> getObjectEvaluation(
+        String key, Value defaultValue, EvaluationContext evaluationContext) {
+        return resolveFlagsmithEvaluation(key, defaultValue, evaluationContext, Object.class);
+    }
+
+    /**
+     * Using the Flagsmith SDK this method resolves any type of flag into
+     * a ProviderEvaluation. Since Flagsmith's sdk is agnostic of type
+     * the flag needs to be cast to the correct type for OpenFeature's
+     * ProviderEvaluation object.
+     *
+     * @param key          the string identifier for the flag being resolved
+     * @param defaultValue the backup value if the flag can't be resolved
+     * @param ctx          an EvaluationContext object with flag evaluation options
+     * @param expectedType the expected data type of the flag as a class
+     * @param <T>          the data type of the flag
+     * @return a ProviderEvaluation object for the given flag type
+     * @throws OpenFeatureError
+     */
+    private <T> ProviderEvaluation<T> resolveFlagsmithEvaluation(
+        String key, T defaultValue, EvaluationContext ctx, Class<?> expectedType
+    ) throws OpenFeatureError {
+        T flagValue = null;
+        ErrorCode errorCode = null;
+        Reason reason = null;
+        String variationType = "";
+        try {
+
+            Flags flags = Objects.isNull(ctx.getTargetingKey()) || ctx.getTargetingKey().isEmpty()
+                ? flagsmith.getEnvironmentFlags()
+                : flagsmith.getIdentityFlags(ctx.getTargetingKey());
+            // Check if the flag is enabled, return default value if not
+            if (!flags.isFeatureEnabled(key)) {
+                return ProviderEvaluation.<T>builder()
+                                         .value(defaultValue)
+                                         .reason(Reason.DISABLED.name())
+                                         .build();
+            }
+
+            Object value = flags.getFeatureValue(key);
+            // Convert the value received from Flagsmith.
+            flagValue = convertValue(value, expectedType);
+
+            if (flagValue.getClass() != expectedType) {
+                throw new TypeMismatchError("Flag value " + key + " had unexpected type "
+                    + flagValue.getClass() + ", expected " + expectedType + ".");
+            }
+
+        } catch (FlagsmithApiError flagsmithApiError) {
+            flagValue = defaultValue;
+            reason = Reason.ERROR;
+            errorCode = ErrorCode.PARSE_ERROR;
```

Comment:
Not sure why this is a PARSE_ERROR?

Reply:
Agreed it's been changed to a GeneralError